lib/dependency-versions.ts

```diff
@@ -209,12 +209,15 @@ export function fixMismatchingVersions(
             autosave: true,
             stringify_eol: packageJsonEndsInNewline, // If a newline at end of file exists, keep it.
           });
+
           packageJsonEditor.set(
             `devDependencies.${mismatchingVersion.dependency.replace(
               /\./g, // Escape dots.
               '\\.'
             )}`,
-            fixedVersion
+            fixedVersion,
+          // @ts-ignore (@types/edit-json-file not available for 1.7)
+           { preservePaths: false }
```

Comment:
Can you add a comment to this line explaining what problem this [preservePaths](https://github.com/jonschlinkert/set-value#optionspreservepaths) option fixes and how it fixes it?

Reply:
Hi @bmish , sorry I didn't explain it very well. If I ran --fix and the dependency name had a slash in, the resulting package.json had a new entry *outside* of dependencies, which looked like this:

```
"dependencies: { ... },
"devDependencies": { ... },
"devDependencies.@types/jest": "^27.0.2",
```

The fix itself is a setting which is supplied to the set-value library, which is a dependency of edit-json-file, that you are using in your tool. You can see the doc for the preservePaths option here: https://github.com/jonschlinkert/set-value#optionspreservepaths - without this setting it assumes that if the property includes a /, it must be a url, so shouldn't be split even on dots. If that makes sense! It took me a while to track down.

Thanks for accepting the PR.

Reply:
That's pretty obscure, great find!